crates/analyzer/src/db/queries/events.rs

```diff
@@ -40,7 +40,17 @@ pub fn event_type(db: &dyn AnalyzerDb, event: EventId) -> Analysis<Rc<types::Eve
             } = &field.kind;
 
             let typ = type_desc(&mut scope, typ_node).and_then(|typ| match typ {
-                typ if typ.has_fixed_size(scope.db()) => Ok(typ),
+                typ if typ.has_fixed_size(scope.db()) => {
```

Comment:
There is no need to reject all zero-sized fields. It'd be better to reject a field iff its type is zero-sized *and* it is indexed.

Reply:
Done